corehq/apps/geospatial/static/geospatial/js/gps_capture.js

```diff
@@ -46,9 +46,9 @@ hqDefine("geospatial/js/gps_capture",[
             let latNum = parseFloat(self.lat());
             let lonNum = parseFloat(self.lon());
 
-            const latValid = (latNum >= -90 && latNum <= 90) || !self.lat().length;
+            const latValid = (!isNaN(latNum) && latNum >= -90 && latNum <= 90) || !self.lat().length;
```

Comment:
why would latNum not be a number? isn't that guaranteed from parseFloat?

Reply:
Since the input field has been changed from "number" to "text", the user is free to give any string input. If a user were to give a non-numeric input, then `parseFloat` would return as `NaN`. This check is to ensure that we don't allow the user to save the row if a non-numeric input was given.